src/Catel.Core/Catel.Core.Shared/Data/ModelEqualityComparer.cs

```diff
@@ -159,13 +161,7 @@ public override bool Equals(ModelBase x, ModelBase y)
         /// <returns>A hash code for this instance, suitable for use in hashing algorithms and data structures like a hash table.</returns>
         public override int GetHashCode(ModelBase obj)
         {
-            if (obj == null)
-            {
-                return 0;
-            }
-
-            var objType = obj.GetType();
-            return objType.FullName.GetHashCode();
+            return RuntimeHelpers.GetHashCode(obj);
```

Comment:
Hash is always complicated. I think we should change the hashing to always return GetHashCode. The idea I had a long time ago (read more than 6 years ago) was that if you cloned an object, it should have the same hash. I think it's time to let that idea go, so this code is good.